### PR TITLE
Config/flag: medalla as default

### PR DESCRIPTION
**Pull request by terencechain**
Set medalla as default for config. Change list:

* Preserve mainnet config as mainnet config
* Set medalla as default config during `Configure...()`
* Improved a test to not use hardcoded value. The test was failing due to genesis fork version and i just left the improvement as it is

### CODE REVIEW

File: shared/featureconfig/flags.go [hunk 1]
```diff
@@ -500,6 +495,10 @@ var (
 		Usage:  deprecatedUsage,
 		Hidden: true,
 	}
+	deprecatedMedallaTestnet = &cli.BoolFlag{
+		Name:   "medalla",
+		Usage:  deprecatedUsage,
+		Hidden: true}
```
Comment (rauljordan) [inline shared/featureconfig/flags.go:501]:
```suggestion
		Hidden: true,
	}
```

File: beacon-chain/p2p/fork_test.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ func TestDiscv5_AddRetrieveForkEntryENR(t *testing.T) {
 	localNode := enode.NewLocalNode(db, pkey)
 	localNode.Set(forkEntry)
 
-	want, err := helpers.ComputeForkDigest([]byte{0, 0, 0, 0}, genesisValidatorsRoot)
+	want, err := helpers.ComputeForkDigest([]byte{0, 0, 0, 1}, genesisValidatorsRoot)
```
Comment (nisdas) [inline beacon-chain/p2p/fork_test.go:240]: hmm why is this changed ?
